src/utils/fetchXPrices.js

```diff
@@ -0,0 +1,31 @@
+const BigNumber = require('bignumber.js');
+const { fantomWeb3: web3 } = require('./web3');
+const ERC20 = require('../abis/ERC20.json');
+const fBEETSPool = require('../data/fantom/fBeetsPool.json');
+
+import { addressBook } from '../../packages/address-book/address-book';
+const {
+  fantom: {
+    tokens: { BOO, xBOO, SCREAM, xSCREAM, BEETS, fBEETS },
+  },
+} = addressBook;
+
+const getXPrice = async (tokenPrice, tokenAddress, xTokenAddress) => {
+  const tokenContract = new web3.eth.Contract(ERC20, tokenAddress);
+  const xTokenContract = new web3.eth.Contract(ERC20, xTokenAddress);
+
+  const stakedInXPool = new BigNumber(await tokenContract.methods.balanceOf(xTokenAddress).call());
+  const totalXSupply = new BigNumber(await xTokenContract.methods.totalSupply().call());
+
+  return stakedInXPool.times(tokenPrice).dividedBy(totalXSupply).toNumber();
+};
+
+const fetchXPrices = async tokenPrices => {
+  return {
+    xBOO: await getXPrice(tokenPrices.BOO, BOO.address, xBOO.address),
+    xSCREAM: await getXPrice(tokenPrices.SCREAM, SCREAM.address, xSCREAM.address),
+    fBEETS: await getXPrice(tokenPrices.BEETS, fBEETSPool[0].address, fBEETS.address),
```

Comment:
not bad, but few problems
- only for Fantom, so it can't be in general getAmmPrices
- no multicall, instead of 1 request (fetchMooPrices) there are 6 requests.
- all request are sequential, not parallel
- fBEETS should be based on BEETS-FTM LP price, not single BEETS?

Reply:
Very helpful feedback, will have it fixed up tomorrow